### PR TITLE
Create NanoPb code generation scripts

### PR DESCRIPTION
**Pull request by jeremyjiang-dev**
Note: `perf_metric.options`  is not finalized yet and will be addressed in a following PR.

### CODE REVIEW

File: FirebasePerformance/ProtoSupport/proto_generator.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import os
+import os.path
+import re
+import subprocess
+
+OBJC_GENERATOR='nanopb_objc_generator.py'
+
+COPYRIGHT_NOTICE = '''
+/*
+ * Copyright 2021 Google
```
Comment (paulb777) [inline FirebasePerformance/ProtoSupport/proto_generator.py:43]: LLC

**Review by paulb777 — APPROVED** [review]

File: FirebasePerformance/ProtoSupport/proto_generator.py (hunk 1 above)
Comment (jeremyjiang-dev) [reply FirebasePerformance/ProtoSupport/proto_generator.py]: Good catch! Thanks Paul!

**Review by visumickey — APPROVED** [review]

File: FirebasePerformance/ProtoSupport/Protos/perf_metric.options [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Google
```
[the remark this answers is not in the document]
Comment (jeremyjiang-dev) [reply FirebasePerformance/ProtoSupport/Protos/perf_metric.options:1]: Yes. I will merge this change to `perfSPM` and the options file will be finalized in a following PR.

File: FirebasePerformance/ProtoSupport/Protos/perf_metric.options [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+firebase.perf.v1.GaugeMetadata.process_name type:FT_POINTER
+
+firebase.perf.v1.ApplicationInfo.google_app_id type:FT_POINTER
+firebase.perf.v1.ApplicationInfo.app_instance_id type:FT_POINTER
+firebase.perf.v1.ApplicationInfo.custom_attributes type:FT_POINTER
+firebase.perf.v1.ApplicationInfo.Custom_Attributes.key type:FT_POINTER
+firebase.perf.v1.ApplicationInfo.Custom_Attributes.value type:FT_POINTER
+
+firebase.perf.v1.WebApplicationInfo.sdk_version type:FT_POINTER
```
[the remark this answers is not in the document]
Comment (jeremyjiang-dev) [reply FirebasePerformance/ProtoSupport/Protos/perf_metric.options:38]: Sg. I will remove them in a following PR.

File: FirebasePerformance/ProtoSupport/README.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+To build the protos:
+- `brew install protobuf`
+- Verify version in generate_protos.sh
+- `./generate_protos.sh`
```
[the remark this answers is not in the document]
Comment (jeremyjiang-dev) [reply FirebasePerformance/ProtoSupport/README.md:4]: Sg. Updated.

**Pull request merged**